File: RomanToInt.java
// import java.util.*;

// public class RomanToInt {
//      public static int romanToInt(String s) {
//         // Map Roman numerals to their integer values
//         HashMap<Character, Integer> romanToValue = new HashMap<>();
//         romanToValue.put('I', 1);
//         romanToValue.put('V', 5);
//         romanToValue.put('X', 10);
//         romanToValue.put('L', 50);
//         romanToValue.put('C', 100);
//         romanToValue.put('D', 500);
//         romanToValue.put('M', 1000);

//         // Initialize result
//         int total = 0;

//         // Loop through the string
//         for (int i = 0; i < s.length(); i++) {
//             // Get the value of the current symbol
//             int current = romanToValue.get(s.charAt(i));
            
//             // Get the value of the next symbol if it exists
//             if (i < s.length() - 1 && current < romanToValue.get(s.charAt(i + 1))) {
//                 total -= current;
//             } else {
//                 total += current;
//             }
//         }

//         return total;
//     }

//     public static void main(String[] args) {
//         String s = "III";
//         System.out.println(romanToInt(s));  // Output: 3
//     }
// }

File: RomanToInt.py
def roman_to_int(s):
    # Dictionary to map Roman numerals to integers
    roman_to_value = {
        'I': 1, 'V': 5, 'X': 10, 'L': 50,
        'C': 100, 'D': 500, 'M': 1000
    }
    
    # Initialize result
    total = 0
    
    # Loop through the string
    for i in range(len(s)):
        # If the current value is less than the next value, subtract it
        if i < len(s) - 1 and roman_to_value[s[i]] < roman_to_value[s[i + 1]]:
            total -= roman_to_value[s[i]]
        else:
            # Otherwise, add it
            total += roman_to_value[s[i]]
    
    return total

# Example usage
s = "XIII"
print(roman_to_int(s))  # Output: 3